### ISSUE TITLE
trace页面 ui服务报错： NullPointerException 

### ISSUE THREAD

**Issue by VectorHo** [wu-sheng/sky-walking#203]
## 前端报错页面：

![image](https://cloud.githubusercontent.com/assets/7578264/26677423/7afb16a4-46fe-11e7-93bf-e13ba0a89db4.png)

## 对应后端报错：
2017-06-01 11:10:33,213 - com.a.eye.skywalking.ui.controller.TopTraceListController -168707 [qtp2051598050-13] DEBUG [] - topTraceListDataLoad startTime = 201705300000, endTime = 201706019999, from=0, minCost=-1, maxCost=-1, globalTraceId=
2017-06-01 11:10:33,213 - com.a.eye.skywalking.ui.controller.TopTraceListController -168707 [qtp2051598050-13] DEBUG [] - topTraceListDataLoad startTime = 201705300000, endTime = 201706019999, from=0, minCost=-1, maxCost=-1, globalTraceId=
2017-06-01 11:10:33,231 - com.a.eye.skywalking.ui.tools.HttpClientTools -168725 [qtp2051598050-13] DEBUG [] - executing get request http://skywalking-collector:12800/segments/top/timeSlice?startTime=201705300000&endTime=201706019999&from=0&limit=30&minCost=-1&maxCost=-1
2017-06-01 11:10:33,231 - com.a.eye.skywalking.ui.tools.HttpClientTools -168725 [qtp2051598050-13] DEBUG [] - executing get request http://skywalking-collector:12800/segments/top/timeSlice?startTime=201705300000&endTime=201706019999&from=0&limit=30&minCost=-1&maxCost=-1
2017-06-01 11:10:33,270 - com.a.eye.skywalking.ui.service.TopTraceListService -168764 [qtp2051598050-13] DEBUG [] - load top segment data: {"isSuccess":false,"reason":null}
2017-06-01 11:10:33,270 - com.a.eye.skywalking.ui.service.TopTraceListService -168764 [qtp2051598050-13] DEBUG [] - load top segment data: {"isSuccess":false,"reason":null}
2017-06-01 11:10:33,271 - org.eclipse.jetty.server.HttpChannel -168765 [qtp2051598050-13] WARN  [] - /topTraceListDataLoad
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is java.lang.NullPointerException
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:982) ~[spring-webmvc-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861) ~[spring-webmvc-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687) ~[javax.servlet-api-3.1.0.jar:3.1.0]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846) ~[spring-webmvc-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[javax.servlet-api-3.1.0.jar:3.1.0]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:841) ~[jetty-servlet-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634) ~[jetty-servlet-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:206) ~[websocket-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1621) ~[jetty-servlet-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1621) ~[jetty-servlet-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.springframework.web.filter.HttpPutFormContentFilter.doFilterInternal(HttpPutFormContentFilter.java:105) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1621) ~[jetty-servlet-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:81) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1621) ~[jetty-servlet-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1621) ~[jetty-servlet-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:541) ~[jetty-servlet-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548) ~[jetty-security-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1592) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1239) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:481) ~[jetty-servlet-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1561) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1141) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.Server.handle(Server.java:564) ~[jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:320) [jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251) [jetty-server-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279) [jetty-io-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:110) [jetty-io-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124) [jetty-io-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:672) [jetty-util-9.4.2.v20170220.jar:9.4.2.v20170220]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:590) [jetty-util-9.4.2.v20170220.jar:9.4.2.v20170220]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException
	at 

**com.a.eye.skywalking.ui.service.TopTraceListService.topTraceListDataLoad(TopTraceListService.java:98) ~[sky-walking-ui-3.0.1-2017.jar:?]
	at com.a.eye.skywalking.ui.controller.TopTraceListController.topTraceListDataLoad(TopTraceListController.java:36) ~[sky-walking-ui-3.0.1-2017.jar:?]**

at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:133) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:116) ~[spring-webmvc-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827) ~[spring-webmvc-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738) ~[spring-webmvc-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85) ~[spring-webmvc-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963) ~[spring-webmvc-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:897) ~[spring-webmvc-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970) ~[spring-webmvc-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	... 44 more

**Comment by VectorHo** [comment]
找了一下报错好像是这个条：

**Comment by pengys5** [comment]
后台有错, load top segment data: {"isSuccess":false,"reason":null}
看看Collector的日志呗.

**Comment by VectorHo** [comment]
@pengys5 好的，我去掉一个工作流相关的服务就正常了，工作流服务就用了activiti，pom配置如下：
![9772c95f9b8221e47943464d063a880f](https://cloud.githubusercontent.com/assets/7578264/26708619/76bf8f80-477f-11e7-90d6-b5a19580a5a2.png)
![41d9e8b6e7c6380e71a89b0bf5eb84fd](https://cloud.githubusercontent.com/assets/7578264/26708623/7b2b58ec-477f-11e7-8f86-5e798468b7eb.png)

**Comment by VectorHo** [comment]
@pengys5 ，这个日志如下：
![image](https://cloud.githubusercontent.com/assets/7578264/26708769/7d1fa792-4780-11e7-9c1c-5c6c69ca31c6.png)

### Collector：skywalking-server-log4j2.log 报错日志

```java
2017-06-02 02:48:43,571 - com.a.eye.skywalking.collector.worker.storage.PersistenceTimer -54266493 [timerPersistence] INFO  [] - worker role name: SegmentExceptionSave
2017-06-02 02:48:43,571 - com.a.eye.skywalking.collector.worker.storage.PersistenceTimer -54266493 [timerPersistence] INFO  [] - worker role name: SegmentExceptionSave
2017-06-02 02:48:43,576 - com.a.eye.skywalking.collector.worker.storage.EsClient -54266498 [timerPersistence] INFO  [] - bulk data size: 23
2017-06-02 02:48:43,576 - com.a.eye.skywalking.collector.worker.storage.EsClient -54266498 [timerPersistence] INFO  [] - bulk data size: 23
2017-06-02 02:48:43,725 - com.a.eye.skywalking.collector.worker.storage.EsClient -54266647 [timerPersistence] ERROR [] - failure in bulk execution:
[11]: index [segment_idx], type [record], id [Segment.1496371722772.-1247510313.1.41.45], message [MapperParsingException[failed to parse [ss.ts.db.statement]]; nested: IOException[Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
 at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper@515d2892; line: 1, column: 364]];]
[14]: index [segment_idx], type [record], id [Segment.1496371722862.-1247510313.1.42.45], message [MapperParsingException[failed to parse [ss.ts.db.statement]]; nested: IOException[Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
 at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper@153f4ad2; line: 1, column: 364]];]
2017-06-02 02:48:43,725 - com.a.eye.skywalking.collector.worker.storage.EsClient -54266647 [timerPersistence] ERROR [] - failure in bulk execution:
[11]: index [segment_idx], type [record], id [Segment.1496371722772.-1247510313.1.41.45], message [MapperParsingException[failed to parse [ss.ts.db.statement]]; nested: IOException[Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
 at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper@515d2892; line: 1, column: 364]];]
[14]: index [segment_idx], type [record], id [Segment.1496371722862.-1247510313.1.42.45], message [MapperParsingException[failed to parse [ss.ts.db.statement]]; nested: IOException[Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
 at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper@153f4ad2; line: 1, column: 364]];]
2017-06-02 02:48:46,726 - com.a.eye.skywalking.collector.worker.storage.PersistenceTimer -54269648 [timerPersistence] INFO  [] - worker role name: GlobalTraceSave
2017-06-02 02:48:46,726 - com.a.eye.skywalking.collector.worker.storage.PersistenceTimer -54269648 [timerPersistence] INFO  [] - worker role name: GlobalTraceSave
```

### 日志内异常报告
![image](https://cloud.githubusercontent.com/assets/7578264/26708908/5e9085fc-4781-11e7-951a-a7cf5197d8c7.png)


### btw
skywalking都是跑在docker容器的

**Comment by wu-sheng** [comment]
@pengys5 还是json的string处理有问题。上次发现的BUG。

ref: https://github.com/wu-sheng/sky-walking/issues/200

**Comment by pengys5** [comment]
嗯, 看到了, json 特殊字符未转义直接入es的问题, 今天晚上修复.

**Comment by VectorHo** [comment]
@pengys5 好的，修复好了请说一下；我们这个很急用 T_T

**Comment by wu-sheng** [comment]
@VectorHo 能介绍一下是在哪个公司的哪个项目里面用么？

**Comment by VectorHo** [comment]
@wu-sheng 主要是我们自己的微服务平台www.moria.io，其中微服务业务开发部分java开发框架是用到了dubbox和springboot，对比了很多分布式调用链；skywalking比较合适

**Comment by wu-sheng** [comment]
@VectorHo 在线服务是我们很感兴趣的，进行支持的领域。有问题及时沟通。

**Comment by VectorHo** [comment]
@wu-sheng @pengys5 期待你们的产品更加完善

**Comment by wu-sheng** [comment]
@VectorHo 能够提供给我和 @pengys5 更直接的联系方式，比如QQ或者微信。想更多的了解你们的部署和使用模式。

**Comment by VectorHo** [comment]
@wu-sheng 微信号：heyanjiemao

**Comment by VectorHo** [comment]
@wu-sheng 这个fix是跟着哪个版本上？我暂时用docker-compose的方式适用skywalking

**Comment by wu-sheng** [comment]
跟在3.0.1上，发布3.0.2，做一个快速修复。然后3.1会做一个更彻底的改造。

**Comment by wu-sheng** [comment]
Fixed by #207
